ens.asciidoc

```diff
@@ -13,31 +18,75 @@ The _Ethereum Name Service (ENS)_ is one of the first Ethereum DApps, offering a
+
+=== Bottom of the sandwich: Name Owners and resolvers
+
+The base layer of the sandwich is a cleverly simple (less than 50 lines of code) contract defined by ERC137 that basically only allows owners of names to set information about their names and to create subnodes (subdomains). The core of the simplicity is the identifier mechanism for names, called namehash.
+
+Namehash is an algorithm for converting any name in a identifier hash of the name, and it works like this:
+
+Root Node:  `keccak(0x0000000000000000000000000000000000000000000000000000000000000000)`
+".eth" node: `keccak( (Root Node) + keccak('eth'))`
+"example.eth" node: `keccak( ('.eth' node) + keccak('example'))`
+"subdomain.example.eth" node: `keccak( ('example.eth' node) + keccak('subdomain'))`
+
+And recursively so. The only functions on the base layer are those that enable a node owner to set information about their own node (specifically: resolver, time to live and transferring ownership) and to create an owner to a subnode.
+
+==== Rootnode ownership
+
+One of the results of this hierarchical system is that it relies on the owners of the Rootnode, that are able to create top level domains. While the eventual goal is to move this into some sort of decentralized decision making process for new top level domains, right now the rootnode is a 4 out of 7 multisig, held by people in different countries (built as a reflection of the seven keyholders of the DNS system). Currently the purpose and goal of these keyholders is to work in consensus with the community to:
+
+* Migrate and upgrade the temporary ownrship of the ".eth" TLD to a more permanent contract once the system is evaluated
+* Allow adding new TLDs, if the community agrees that such is ever needed
+* Migrate the ownership of the root multisig in a more decentralized contract, when such system is agreed upon, tested and implemented.
+* To serve as a last resort way to deal with any bugs or vulnerability on the top level registries
+
+==== Resolvers
+
+The basic ENS contract doesn't allow owners of adding any metadata to contracts, that is the job of the resolver contracts. The purpose is to allow more flexibility by allowing users to create a custom contract that can answer any questions about the name in itself, like "what's the swarm address associated with this app?", "what is the ethereum address that receives payments in ether and tokens?", "what's the hash of the app, in order to verify its integrity?".
+
+=== Middle layer: the ".eth" nodes
+
+At the moment, the only top level domain that is uniquely registerable in a smart contract is ".eth". There's work on enabling traditional dns domain owners to claim ENS ownership. While in theory this could work for ".com" the only domain that this has been [implemented is for ".xyz" in a testnet](https://medium.com/the-ethereum-name-service/how-to-claim-your-dns-domain-on-ens-e600ef2d92ca).
```

Comment:
Could it be **"registrable"** rather than **"registerable"** or maybe it is equally valid?